src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Configuration/ApplicationInsightsSinkRequestOptions.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using GuardNet;
+
+namespace Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights.Configuration
+{
+    /// <summary>
+    /// User-defined configuration options to influence the behavior of the Azure Application Insights Serilog sink while tracking requests.
+    /// </summary>
+    public class ApplicationInsightsSinkRequestOptions
+    {
+        private Func<string> _generatedId = Guid.NewGuid().ToString;
+
+        /// <summary>
+        /// <para>Gets or sets the function to generate the request ID of the telemetry model while tracking requests.</para>
+        /// <para>Default: GUID generation.</para>
+        /// </summary>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="value"/> is <c>null</c>.</exception>
+        public Func<string> GenerateId
```

Comment:
Would a name like `IdGenerator` be more appropriate for this property ?  Maybe it is more readable when configuring:

```
options => {
    options.IdGenerator = () => .....
}
```

Reply:
Hmm, I like to use verbs for functions and methods. We already use `GenerateId` for the correlation info so to be consistent, it seemed appropriate to keep this naming.